torchx/c_src/torchx.cpp

```diff
@@ -277,12 +277,18 @@ NIF(to_blob)
   {
     return nx::nif::ok(env, enif_make_resource_binary(env, t, data_ptr, byte_size));
   }
-  else
+  else if (device.has_value() && device.value().type() == torch::kCPU)
   {
     void *result_data = (void *)enif_make_new_binary(env, byte_size, &result);
     memcpy(result_data, data_ptr, byte_size);
     return nx::nif::ok(env, result);
   }
+  else
+  {
+    void *result_data = (void *)enif_make_new_binary(env, byte_size, &result);
+    memcpy(result_data, reshaped.to(torch::kCPU).data_ptr(), byte_size);
```

Comment:
This will transfer everything from the GPU, even if we are going to read only part of it. What if we trim down the tensor and make sure it has at most `limit` elements? We should also probably raise if `limit` is more than the number of elements?

Reply:
We could take a slice of the tensor given the limit. I wouldn't raise if limit is more than the number of elements, though, because it is just an upper bound (for instance, `limit: :infinity` wouldn't make sense if we only accepted a limit within the tensor bounds)